### ISSUE TITLE
wallet listen --port 3416 still tries to listen on 3415

### ISSUE THREAD

**Issue by aleqx** [mimblewimble/grin#2893]
I'm using grin 1.0.3 (release) on Ubuntu 16.04.5 LTS. I have successfully launched one wallet with `grin wallet listen` which listens on port 3415 for account `default` and it works fine.

I then did `grin wallet account -c grin2` to create a second account, which worked fine, and then I wanted to launch another listener on port 3416 for this account, but it fails complaining it's already listening on 3415 ... not 3416. 

It seems it ignores the `--port 3416` option and still tries to listen on 3415.

```
# grin wallet -a grin2 listen --port 3416
Password:
20190612 14:25:43.143 WARN grin_wallet::controller - Starting HTTP Foreign listener API server at 0.0.0.0:3415.
20190612 14:25:43.144 WARN grin_wallet::controller - HTTP Foreign listener started.
20190612 14:25:43.207 ERROR grin_util::logger -
thread 'apis' panicked at 'error binding to 0.0.0.0:3415: error creating server listener: Address already in use (os error 98)': /root/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.12.19/src/server/mod.rs:115stack backtrace:
   0: grin_util::logger::send_panic_to_log::{{closure}}::hda887ed4c3c0301f (0x55c9423807b0)
   1: rust_panic_with_hook (0x55c9424c2ae8)
             at src/libstd/panicking.rs:478
   2: continue_panic_fmt (0x55c9424c2581)
             at src/libstd/panicking.rs:381
   3: begin_panic_fmt (0x55c9424c24ce)
             at src/libstd/panicking.rs:336
   4: hyper::server::Server<hyper::server::tcp::AddrIncoming,()>::bind::h55d84a33235c1d73 (0x55c94213454a)
   5: std::sys_common::backtrace::__rust_begin_short_backtrace::h0c04b1b57af607a4 (0x55c941fa3353)
   6: std::panicking::try::do_call::h9b84fbb39fb00c4e (0x55c94206267d)
   7: __rust_maybe_catch_panic (0x55c9424cc929)
             at src/libpanic_unwind/lib.rs:87
   8: core::ops::function::FnOnce::call_once{{vtable.shim}}::h794cbfc277aff8c6 (0x55c941f611e2)
   9: call_once<(),FnBox<()>> (0x55c9424aebce)
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/liballoc/boxed.rs:702
  10: call_once<(),alloc::boxed::Box<FnBox<()>>> (0x55c9424cbcbf)
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/liballoc/boxed.rs:702
      start_thread
             at src/libstd/sys_common/thread.rs:14
      thread_start
             at src/libstd/sys/unix/thread.rs:80
  11: start_thread (0x7fec4a60d6b9)
  12: clone (0x7fec4a12d41c)
  13: <unknown> (0x0)


Thread 'apis' panicked with message:
"error binding to 0.0.0.0:3415: error creating server listener: Address already in use (os error 98)"
See /root/.grin/main/grin-wallet.log for further details.
Wallet command failed: LibWallet Error: Unknown
```

**Comment by aleqx** [comment]
Killing the existing default listener and trying again, it listens on 3415 instead of 3416

```
# grin wallet -a grin2 listen --port 3416
Password:
20190612 14:36:51.281 WARN grin_wallet::controller - Starting HTTP Foreign listener API server at 0.0.0.0:3415.
20190612 14:36:51.282 WARN grin_wallet::controller - HTTP Foreign listener started.
```

**Comment by aleqx** [comment]
I see this is fixed in the new grin-wallet 1.1.0

**Comment by aleqx** [comment]
Actually, it would be good to fix it for the 1.0.x branch as well, because lots of people are still running it while waiting for 1.1 to become stable.

**Comment by hashmap** [comment]
Closing because of 1.0 sunset
